src/twisted/web/http.py

```diff
@@ -1516,6 +1516,15 @@ def __ne__(self, other):
         return NotImplemented
 
 
+    def __hash__(self):
+        """
+        A C{Request} is hashable so that it can be used as a mapping key.
+
+        @return A C{int} based on the instance's identity.
+        """
+        return id(self)
```

Comment:
This is OK under PyPy (see review comment) and fine otherwise.

Please add this to the `_IDeprecatedHTTPChannelToRequestInterface`.